dxr/plugins/extmatch.py

```diff
@@ -0,0 +1,73 @@
+"""Extmatch - Match files in a given folder by extension and provide a link
+in the nav panel of each to the other.
+
+Currently we match "Header" file extensions with "Implementation" file
+extensions.
+"""
+
+from fnmatch import fnmatchcase
+from os.path import relpath, join, basename, splitext, isfile
+from collections import namedtuple
+
+from dxr.indexers import (FileToIndex as FileToIndexBase,
+                          TreeToIndex as TreeToIndexBase)
+from dxr.mime import icon
+from dxr.utils import browse_file_url
+
+# A tuple of extensions along with a title describing the type of extensions:
+_TitledExts = namedtuple('_TitledExts', ['exts', 'title'])
+
+
+class TreeToIndex(TreeToIndexBase):
+    def __init__(self, plugin_name, tree, vcs_cache):
+        super(TreeToIndex, self).__init__(plugin_name, tree, vcs_cache)
+        self.header_exts = _TitledExts(('.h', '.hpp', '.hxx'), 'Header')
```

Comment:
A nit: we should use lists rather than tuples for the `exts` value. In DXR, we've do far followed the convention that tuples are for anonymous structs and lists are for homogeneous sequences of things. Quite possibly we should use something more efficient instead, since we're doing a lot of linear scanning over them using `in`. But the data sizes are small, so we might not see a benefit.